common/djangoapps/student/helpers.py

```diff
@@ -264,46 +272,61 @@ def get_next_url_for_login_page(request):
         # Don't add tpa_hint if we're already in the TPA pipeline (prevent infinite loop),
         # and don't overwrite any existing tpa_hint params (allow tpa_hint override).
         running_pipeline = third_party_auth.pipeline.get(request)
-        (scheme, netloc, path, query, fragment) = list(six.moves.urllib.parse.urlsplit(redirect_to))
+        (scheme, netloc, path, query, fragment) = list(urllib.parse.urlsplit(redirect_to))
         if not running_pipeline and 'tpa_hint' not in query:
-            params = six.moves.urllib.parse.parse_qs(query)
+            params = urllib.parse.parse_qs(query)
             params['tpa_hint'] = [tpa_hint]
-            query = six.moves.urllib.parse.urlencode(params, doseq=True)
-            redirect_to = six.moves.urllib.parse.urlunsplit((scheme, netloc, path, query, fragment))
+            query = urllib.parse.urlencode(params, doseq=True)
+            redirect_to = urllib.parse.urlunsplit((scheme, netloc, path, query, fragment))
 
     return redirect_to
 
 
-def _get_redirect_to(request):
+def _get_redirect_to(request_host, request_headers, request_params, request_is_https):
```

Comment:
Note for reviewers: I exploded request into four arguments so that this function is agnostic to whether it is a POST or GET request.